test/inductor/test_auto_functionalize.py

```diff
@@ -964,6 +966,219 @@ def f(x):
             # to clone not-inplaced args.
             f(x[1])
 
+    @torch._inductor.config.patch(enable_auto_functionalized_v2=True)
+    def test_alias(self, _dynamic=False):
+        with torch.library._scoped_library("mylib", "FRAGMENT") as lib:
+            torch.library.define(
+                "mylib::foo",
+                "(Tensor(a!) x, Tensor(b!) y) -> ()",
+                tags=torch.Tag.pt2_compliant_tag,
+                lib=lib,
+            )
+
+            @torch.library.impl("mylib::foo", "cpu", lib=lib)
+            @torch._dynamo.disable
+            def foo_impl(x, y):
+                x.sin_()
+                y.sin_()
+
+            def f(x):
+                a = torch.ops.aten.alias.default(x)
```

Comment:
can you make this something like...
```py
def f(x):
    a = torch.ops.aten.alias.default(x)
    b = x.clone()
    c = b.nonzero().float()
    d = c.alias()
    torch.ops.mylib.foo(a, d)
    return a, d
```
`d` is a Tensor with unbacked symints in the shape

Reply:
i will add a test for that i here and in the recompile test

Reply:
added they pass